### ISSUE TITLE
Pre-created test-database for actor and protocol tests

### ISSUE THREAD

**Issue by da-kami** [itchysats/itchysats#2928]
## Problem Statement

At the moment we always setup up the state of the system by actually performing all tasks in the system to achieve a certain state. E.g. in order to perform a rollover we actually run a contract setup between the two parties and then trigger the rollover once we created the `Cfd`.

Note that such a setup does not only concern the actor tests, but is also very useful for protocol tests. For don't have tests for the rollover and collab settlement protocol at the moment because of a lack of having a database that provides the necessary state.
The more protocol tests we can write in isolation the less more complex actor tests for the whole system we need.

Our current actor test setup is expensive and can be optimized. Instead of creating the state to perform a task we can just provide the state. There are different ways to achieve this.

## Solution Proposal

I propose to have a separate `test-data` crate in the `itchysats` repo that provides specific databases (one for the maker and one for the taker) that can then used in the actor tests and protocol tests. This database files are checked into the repository and used other crates wherever we need it (I think we should be able to achieve that with simple relative paths as long as we keep this within the same repo; for a more complex setup we could actually link to the repo). Whenever changes to the test data are needed we update the database.

A specific db filled with test data has the advantage that we don't have to do any setup in the test prior to running it, but can just load the `Cfd` and test certain tasks on top of the state in the db. 
It has the disadvantage that when we want to extend / change the test database setup we cannot just do that in the test, but have to update the database - or potentially go further and create multiple, versioned databases (initially I would not do that).

Initially it will be sufficient to provide a database that contains one `Cfd`that is `Open`. This `Cfd` will always be the same and is not configurable. This is a quite rigid yet simple setup.
A test depending on the test database creates a copy (file copy) of the database and then use it; since tests run in parallel we should never access the same database from multiple tests.

Eventually we can change this setup to allow having multiple `Cfd`s in the database. When running a test we can then:

1. Create a copy of the database
2. Delete all `Cfd`s that we are not interested in (i.e. delete all `Cfd`s besides id `xxx` that the test will load). - delete is cheap
3. Use the database for the test setup
4. Run the test

**Comment by da-kami** [comment]
Note: 
I did think about a setup where, instead of using a pre-defined database, we would have e.g. JSON data that we write into the database when setting up each test. But I did not see an advantage in that. We cannot really modify the JSON because that would just get messy, we would have to re-generate it. Then we might as well just re-generate the database instead of loosing time writing the data to the database in each test setup.

As far as I know rust's test framework does not have a `setup` and `tear-down` functionality that can be run for a complete test suite. That would come in handy, because then we could create the database for all tests before running the tests and would not have to check it into the repo. We could work around that by creating a test setup that is not just pure rust, but triggers creation of the test-data database first - That would be a more dynamic setup. I don't see advantages in that for now, because it adds complexity but not many advantage over checking the database into the repo.
